src/app/frontend/typings/backendapi.ts

```diff
@@ -1216,17 +1225,105 @@ export interface SystemBanner {
   severity: string;
 }
 
+// https://godoc.org/k8s.io/api/core/v1#Volumes
 export interface PersistentVolumeSource {
-  gcePersistentDisk: GCEPersistentDiskVolumeSource;
-  awsElasticBlockStore: AWSElasticBlockStorageVolumeSource;
-  hostPath: HostPathVolumeSource;
-  glusterfs: GlusterfsVolumeSource;
-  nfs: NFSVolumeSource;
-  rbd: RBDVolumeSource;
-  iscsi: ISCSIVolumeSource;
-  cinder: CinderVolumeSource;
-  fc: FCVolumeSource;
-  flocker: FlockerVolumeSource;
+  name: string;
+  hostPath?: HostPathVolumeSource;
+  emptyDir?: {};
+  gcePersistentDisk?: GCEPersistentDiskVolumeSource;
+  awsElasticBlockStore?: AWSElasticBlockStorageVolumeSource;
+  gitRepo?: GitRepoVolumeSource;
+  secret?: SecretVolumeSource;
+  nfs?: NFSVolumeSource;
+  iscsi?: ISCSIVolumeSource;
+  glusterfs?: GlusterfsVolumeSource;
+  persistentVolumeClaim?: PersistentVolumeClaimVolumeSource;
+  rbd?: RBDVolumeSource;
+  flexVolume?: FlexVolumeSource;
+  cinder?: CinderVolumeSource;
+  cephFS?: CephFSVolumeSource;
+  flocker?: FlockerVolumeSource;
+  downwardAPI?: DownwardAPIVolumeSource;
+  fc?: FCVolumeSource;
+  azureFile?: AzureFileVolumeSource;
+  configMap?: ConfigMapVolumeSource;
+  vsphereVolume?: VsphereVirtualDiskVolumeSource;
+  quobyte?: QuobyteVolumeSource;
+  // azureDisk: AzureDiskVolumeSource;
```

Comment:
What about this commented code?

Reply:
I've been adding missing types all morning. I was hoping I could finish them before you noticed. Anyway, the code has been pushed and hence we have all the types now!